machinery-helper/tar.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2015 SUSE LLC
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of version 3 of the GNU General Public License as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE about this file by physical or electronic mail,
+// you may find current contact information at www.suse.com
+
+package main
+
+import (
+  "flag"
+  "os"
+  "io"
+  "bufio"
+  "strings"
+  "archive/tar"
+  "compress/gzip"
+  "fmt"
+  "path/filepath"
+  "os/user"
+  "strconv"
+  "os/exec"
+)
+
+type stringArrayFlag []string
+func (i *stringArrayFlag) String() string {
+  return "exclude"
+}
+func (i *stringArrayFlag) Set(value string) error {
+  *i = append(*i, value)
+  return nil
+}
+
+var exclude_list map[string]bool = make(map[string]bool)
```

Comment:
don't use underscores in Go names; var exclude_list should be excludeList<br>should omit type map[string]bool from declaration of var exclude_list; it will be inferred from the right-hand side